ospfd/ospf_vty.c

```diff
@@ -8346,20 +8346,21 @@ DEFUN_HIDDEN (no_ospf_hello_interval,
 	return no_ip_ospf_hello_interval(self, vty, argc, argv);
 }
 
-DEFUN (ip_ospf_network,
-       ip_ospf_network_cmd,
-       "ip ospf network <broadcast|non-broadcast|point-to-multipoint|point-to-point>",
-       "IP Information\n"
-       "OSPF interface commands\n"
-       "Network type\n"
-       "Specify OSPF broadcast multi-access network\n"
-       "Specify OSPF NBMA network\n"
-       "Specify OSPF point-to-multipoint network\n"
-       "Specify OSPF point-to-point network\n")
+DEFUN(ip_ospf_network, ip_ospf_network_cmd,
+      "ip ospf network <broadcast|non-broadcast|point-to-multipoint|point-to-point [dmvpn]>",
+      "IP Information\n"
+      "OSPF interface commands\n"
```

Comment:
@pguibert6WIND, I cannot see any existing yang model for nhrp

Reply:
I had a quick look at the existing yang model, and it is not immediately clear how to add this in. I am not familiar with yang, and I don't know how I would test any changes I made with this.

Reply:
I apologise. actually, ospf is not ready yet for migrating to northbound.
however, to clarify: what kind of network should we configure when we deal with ospf shortcuts ? are we still in 'ip ospf ptp dmvpn' ?

Answering this will help clarify when we move to yang model. one possiblity would be to make a leaf independent, that could be set whatever the ospf network type.

+      leaf dmvpn {
+        type boolean;
+        default "false";
+        description
+          "Used with point-to-point networks.";
+      }
+

Reply:
When using OSPF with DMVPN you need to set it up as a point-to-point at a spoke. However the hub will be setup as point-to-multipoint, and so this would not normally work. The 'point-to-point dmvpn' option makes a spoke send packets that will work correctly in this scenario.

I believe this is actually a subtle violation of the RFC for OSPF, but it is needed to make this work.

Note also that Cisco routers don't have this issue, because in that case you configure the GRE tunnel with a subnet, and setup ospf as 'network broadcast' at both hub and spoke. However in nhrpd we must set the GRE tunnel to a /32 point-point configuration.

Reply:
yang changes will be  done in separate pull request